plugins/tech-insights/src/components/ScorecardsOverview/ScorecardsOverview.tsx

```diff
@@ -32,8 +32,9 @@ export const ScorecardsOverview = ({
 }) => {
   const api = useApi(techInsightsApiRef);
   const { namespace, kind, name } = useParams();
+  const doc = { namespace: namespace!, kind: kind!, name: name! };
```

Comment:
Ah this one was bad to begin with. Could you remove the `useParams` hook usage here and instead do `const { entity } = useEntity()` from the package `@backstage/plugin-catalog-react`? Then you can pass in `getCompoundEntityRef(entity)` from `@backstage/catalog-model` below.

Reply:
Ahhhhh. Thanks! That is much better.

Reply:
Actually no. In fact, these components have no business trying to parse the url at all. The way we've set things up is that there is an entity context around the entire entity page where this should be placed. That way it only has to be loaded once and everything can access that instance freely